server/src/main/java/org/elasticsearch/common/util/CollectionUtils.java

```diff
@@ -211,6 +212,31 @@ public static <E> List<E> appendToCopy(Collection<E> collection, E element) {
         return Collections.unmodifiableList(Arrays.asList(array));
     }
 
+    /**
+     * Creates a copy of the given set and adds extra element.
+     *
+     * @param set     set to copy
+     * @param element element to add
+     */
+    public static <E> Set<E> addToCopy(Set<E> set, E element) {
```

Comment:
Does it make sense to move these methods to `Sets`? `CollectionUtils` already has methods for appending an element to a list which be confusing if have similar methods for sets in the same class but with a different name.